strawberry/cli/__init__.py

```diff
@@ -1,11 +1,15 @@
-from .commands.codegen import codegen as codegen  # noqa
-from .commands.export_schema import export_schema as export_schema  # noqa
-from .commands.server import server as server  # noqa
-from .commands.upgrade import upgrade as upgrade  # noqa
-from .commands.schema_codegen import schema_codegen as schema_codegen  # noqa
+try:
+    from .app import app
+    from .commands.codegen import codegen as codegen  # noqa
+    from .commands.export_schema import export_schema as export_schema  # noqa
+    from .commands.schema_codegen import schema_codegen as schema_codegen  # noqa
+    from .commands.server import server as server  # noqa
+    from .commands.upgrade import upgrade as upgrade  # noqa
 
-from .app import app
+    def run() -> None:
+        app()
 
+except ModuleNotFoundError as exc:
+    from strawberry.exceptions import MissingOptionalDependenciesError
 
-def run() -> None:
-    app()
+    raise MissingOptionalDependenciesError(extras=["cli"]) from exc
```

Comment:
that's great, it's something I was thinking about a few weeks ago!

Do you think it would be possible to make this granular too? Maybe it's not worth it, but some commands need different dependencies (for example schema codegen needs libcst, most of the other commands don't), so maybe we can suggest to install `libcst` or `[cli]` in that case?

But I don't know if we can do this cleanly 🤔

Reply:
Oh wow, how convenient!

If you did want to do that, the cleanest way would probably be to import anything necessary in the commands themselves and raise errors within that. That would add quite a few lines (unless you went to the trouble of using a decorator to transform the error or something), but would get extra granularity when needed.

Idk how worth it it would be though honestly, unless `libcst` installs like 20 dependencies of its own.

I did do it like that for a personal project of mine, where you could use it without the dependencies, but would need them for extra functionality: https://github.com/parafoxia/analytix/blob/main/analytix/reports/interfaces.py#L315-L318

Reply:
yeah, maybe it's not worth doing it for now

I want to add some more commands in future, maybe I can do that then 😊

I'll properly review this and merge later! thanks :)